### PR TITLE
Preload the next image in the background

### PR DESCRIPTION
**Pull request by ulteq**
This preloads the next image in the filelist into the imlib cache.

Closes: #103

- [x] Make it optional (compile flag or feh option)

This results in much faster toggling between images when combined with an increased `--cache-size`.

### CODE REVIEW

File: src/slideshow.c [hunk 1]
```diff
@@ -244,9 +246,26 @@ void feh_reload_image(winwidget w, int resize, int force_new)
 	return;
 }
 
+static void *thread_start(void *arg)
+{
+	feh_file *file = (feh_file*)arg;
+
+	// Hack to prevent segfault when scrolling quickly through already cached images
+	nanosleep((const struct timespec[]){{0, 1000000L}}, NULL);
```
Comment (Ferada) [inline src/slideshow.c]: Why is that necessary, there should probably a better fix available?

Comment (ulteq) [reply src/slideshow.c]: I don't know to be honest. I guess imlib2 is not thread safe.

Comment (Ferada) [reply src/slideshow.c]: Yeah it absolutely is not thread safe. Even if there's a lock around this one instance, all other imlib operations would also need it. Or maybe put all of them into a separate worker thread. In any case, that's all a bit more work.

Comment (ulteq) [reply src/slideshow.c]: I don't think that we can work around this. Proper preloading in the background will always require concurrent imlib calls.

**Comment by ulteq** [comment]
@derf Please let me know what you think about it. And do not hesitate to reject it if you believe it does not fit.

**Comment by derf** [comment]
I like the idea, but won't merge code with possible race conditions -- even if it works on your system, there will probably be environments where the sleep fix doesn't work out.

I'm not sure if synchronizing each and every Imlib2 call in feh is worth the effort though -- perhaps its best to just keep it out of the main feh repository for now.

**Comment by ulteq** [comment]
Sounds reasonable.
